pkg/executor/chained_task_handle.go

```diff
@@ -0,0 +1,112 @@
+package executor
+
+import (
+	"sync"
+	"time"
+)
+
+// ChainedTaskHandle is an links Launchers in a way that
+// one will be launched after another.
+type ChainedTaskHandle struct {
+	TaskHandle
+
+	chainedLauncher Launcher
+
+	encounteredError error
+
+	chainFinished chan struct{}
+	stopChain     chan struct{}
+	stopOnce      sync.Once
+}
+
+// NewChainedTaskHandle returns TaskHandle that executes current handle, and will
+// launch Launcher when handle will finish it's execution.
+func NewChainedTaskHandle(handle TaskHandle, launcher Launcher) TaskHandle {
+	chained := ChainedTaskHandle{
+		TaskHandle:      handle,
+		chainedLauncher: launcher,
+		chainFinished:   make(chan struct{}),
+		stopChain:       make(chan struct{}, 1),
```

Comment:
So that Stop() method will not block whenever the watch() state. In some cases this channel will dangle.